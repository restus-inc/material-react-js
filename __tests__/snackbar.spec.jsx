/* global jest, describe, it, expect */
import React from 'react';
import renderer from 'react-test-renderer';

import Snackbar from '../lib/snackbar';

jest.mock('@material/snackbar');

describe('Snackbar component', () => {
  it('supports snackbar without action button', () => {
    let component = renderer.create(
      <Snackbar message="foo"/>,
    );
    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(
      <Snackbar message="foo" isLeading={true} onClosed={() => {}}/>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('supports snackbar with action button', () => {
    let component = renderer.create(
      <Snackbar label="foo" actionLabel="bar"/>,
    );
    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(
      <Snackbar label="foo" actionLabel="bar" isStacked={true} onClosing={() => {}}/>,
    );
    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(
      <Snackbar label="foo" actionLabel="bar" isLeading={true} onClosed={() => {}}/>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
