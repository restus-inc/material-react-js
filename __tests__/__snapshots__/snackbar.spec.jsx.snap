// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snackbar component supports snackbar with action button 1`] = `
<div
  className="mdc-snackbar"
>
  <div
    className="mdc-snackbar__surface"
  >
    <div
      aria-live="polite"
      className="mdc-snackbar__label"
      role="status"
    >
      foo
    </div>
    <div
      className="mdc-snackbar__actions"
    >
      <button
        className="mdc-button mdc-snackbar__action"
        type="button"
      >
        <div
          className="mdc-button__ripple"
        />
        <span
          className="mdc-button__label"
        >
          bar
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Snackbar component supports snackbar with action button 2`] = `
<div
  className="mdc-snackbar mdc-snackbar--stacked"
>
  <div
    className="mdc-snackbar__surface"
  >
    <div
      aria-live="polite"
      className="mdc-snackbar__label"
      role="status"
    >
      foo
    </div>
    <div
      className="mdc-snackbar__actions"
    >
      <button
        className="mdc-button mdc-snackbar__action"
        type="button"
      >
        <div
          className="mdc-button__ripple"
        />
        <span
          className="mdc-button__label"
        >
          bar
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Snackbar component supports snackbar with action button 3`] = `
<div
  className="mdc-snackbar mdc-snackbar--leading"
>
  <div
    className="mdc-snackbar__surface"
  >
    <div
      aria-live="polite"
      className="mdc-snackbar__label"
      role="status"
    >
      foo
    </div>
    <div
      className="mdc-snackbar__actions"
    >
      <button
        className="mdc-button mdc-snackbar__action"
        type="button"
      >
        <div
          className="mdc-button__ripple"
        />
        <span
          className="mdc-button__label"
        >
          bar
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Snackbar component supports snackbar without action button 1`] = `
<div
  className="mdc-snackbar"
>
  <div
    className="mdc-snackbar__surface"
  >
    <div
      aria-live="polite"
      className="mdc-snackbar__label"
      role="status"
    />
  </div>
</div>
`;

exports[`Snackbar component supports snackbar without action button 2`] = `
<div
  className="mdc-snackbar mdc-snackbar--leading"
>
  <div
    className="mdc-snackbar__surface"
  >
    <div
      aria-live="polite"
      className="mdc-snackbar__label"
      role="status"
    />
  </div>
</div>
`;
